ui/main/src/app/business/view/userCard/userCard.view.spec.ts

```diff
@@ -0,0 +1,849 @@
+/* Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import {getOneCard, initOpfabAPIService, setEntities, setProcessConfiguration, setUserPerimeter} from '@tests/helpers';
+import {ComputedPerimeter} from '@ofModel/userWithPerimeters.model';
+import {RightsEnum} from '@ofModel/perimeter.model';
+import {PermissionEnum} from '@ofModel/permission.model';
+import {UserCardView} from './userCard.view';
+import {UserCardUIControlMock} from '@tests/userCardView/userCardUIControlMock';
+import {EntitiesTree, State} from '@ofModel/processes.model';
+import {EditionMode, InputFieldName} from './userCard.model';
+import {Entity} from '@ofModel/entity.model';
+import {RolesEnum} from '@ofModel/roles.model';
+import {HandlebarsService} from 'app/business/services/card/handlebars.service';
+import {CardAction, Severity} from '@ofModel/light-card.model';
+import {setSpecificCardInformation} from '@tests/userCardView/helpers';
+import {CardServerMock} from '@tests/mocks/cardServer.mock';
+import {CardService} from 'app/business/services/card/card.service';
+import {I18n} from '@ofModel/i18n.model';
+import {CardData} from '@ofModel/card.model';
+import {ServerResponse, ServerResponseStatus} from 'app/business/server/serverResponse';
+
+declare const opfab: any;
+
+describe('UserCard view ', () => {
+    let userCardView: UserCardView;
+    let userCardUIControlMock: UserCardUIControlMock;
+    let cardServerMock: CardServerMock;
+
+    async function initUserCardView(cardWithChildCards?: CardData, editionMode?: EditionMode): Promise<void> {
+        userCardUIControlMock = new UserCardUIControlMock();
+        userCardView = new UserCardView(userCardUIControlMock);
+        setCardDataResponse(cardWithChildCards);
+        await userCardView.init(cardWithChildCards?.card?.id, editionMode);
+    }
+
+    function setCardDataResponse(cardData: CardData) {
+        cardServerMock.setResponseFunctionForLoadCard(() => {
+            return new ServerResponse(cardData, ServerResponseStatus.OK, '');
+        });
+    }
+
+    async function setCurrentUserConfiguration(
+        computedPerimeters: ComputedPerimeter[],
+        permissions: Array<PermissionEnum> = null
+    ): Promise<void> {
```

Comment:
There is no return instruction in this function so maybe the return type should be void (or nothing of course) ?

Reply:
When you have the async keywork , the return type is a Promise 
If it is a Promise<void> you do not need to make return